Source/CarthageKit/Project.swift

```diff
@@ -728,6 +754,37 @@ private func dSYMForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL:
 		|> take(1)
 }
 
+/// Sends the URL to each bcsymbolmap found in the given directory.
+private func BCSymbolMapsInDirectory(directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return filesInDirectory(directoryURL)
+		|> filter { URL in URL.pathExtension == "bcsymbolmap" }
+}
+
+/// Sends the URLs of the bcsymbolmap files that match the given framework and are
+/// located somewhere within the given directory.
+private func BCSymbolMapsForFramework(frameworkURL: NSURL, inDirectoryURL directoryURL: NSURL) -> SignalProducer<NSURL, CarthageError> {
+	return UUIDsForFramework(frameworkURL)
+		|> flatMap(.Merge) { UUIDs in
+			if UUIDs.isEmpty {
+				return SignalProducer.empty
+			}
+			return BCSymbolMapsInDirectory(directoryURL)
+				|> { (signal: Signal<NSURL, CarthageError>) -> Signal<NSURL, CarthageError> in
```

Comment:
I actually find the idea of mapping a signal easier to follow than creating a new `SignalProducer` and starting it immediately. I can extract the logic into a local function if that's easier to read:

```swift
return UUIDsForFramework(frameworkURL)
	|> flatMap(.Merge) { UUIDs in
		if UUIDs.isEmpty {
			return .empty
		}
		func filterUUIDs(signal: Signal<NSURL, CarthageError>) -> Signal<NSURL, CarthageError> {
			var remainingUUIDs = UUIDs
			let count = remainingUUIDs.count
			return signal
				|> filter { fileURL in
					if let basename = fileURL.lastPathComponent?.stringByDeletingPathExtension, fileUUID = NSUUID(UUIDString: basename) {
						return remainingUUIDs.remove(fileUUID) != nil
					} else {
						return false
					}
				}
				|> take(count)
		}
		return BCSymbolMapsInDirectory(directoryURL)
			|> filterUUIDs
}
```